File: youtube-links/src/shared/features/playlists/commands/CreatePlaylist.ts
import { ValidationConsts } from '@/shared/localization/ValidationConsts';

export namespace CreatePlaylist {
  export interface Command {
    name: string;
    public: boolean;
  }

  export const Validation = {
    name: {
      notEmpty: (v: string) => !!v || 'Name should not be empty.',
      maximumLength: (v: string) =>
        v.length <= ValidationConsts.MaximumStringLength ||
        `The length of name must be ${ValidationConsts.MaximumStringLength} characters or fewer. You entered ${v.length} characters.`,
    },
  };
}

File: YoutubeLinks.Shared/Features/Playlists/Commands/CreatePlaylist.cs
using FluentValidation;
using MediatR;
using Microsoft.Extensions.Localization;
using YoutubeLinks.Shared.Localization;

namespace YoutubeLinks.Shared.Features.Playlists.Commands
{
    public static class CreatePlaylist
    {
        public class Command : IRequest<int>
        {
            public string Name { get; set; }
            public bool Public { get; set; }
        }

        public class Validator : AbstractValidator<Command>
        {
            public Validator(IStringLocalizer<ValidationMessage> localizer)
            {
                RuleFor(x => x.Name)
                    .NotEmpty()
                    .WithMessage(x => localizer[nameof(ValidationMessageString.NameNotEmpty)])
                    .MaximumLength(ValidationConsts.MaximumStringLength)
                    .WithMessage(x => localizer[nameof(ValidationMessageString.NameMaximumLength)]);
            }
        }
    }
}
